test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,117 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+
+	"github.com/openshift/origin/test/extended/scheme"
+	"github.com/openshift/origin/test/extended/single_node"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = ginkgo.Describe("[Conformance][sig-sno][Serial] Cluster", func() {
+	f := framework.NewDefaultFramework("cluster-resiliency")
+	f.SkipNamespaceCreation = true
+	f.SkipPrivilegedPSPBinding = true
+
+	oc := exutil.NewCLIWithoutNamespace("cluster-resiliency")
+
+	ginkgo.It("should allow a fast rollout of kube-apiserver", func() {
+		controlPlaneTopology, _ := single_node.GetTopologies(f)
+
+		if controlPlaneTopology != configv1.SingleReplicaTopologyMode {
+			e2eskipper.Skipf("Test is only relevant for single replica topologies")
+		}
+
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+
+		setRESTConfigDefaults(config)
+		restClient, err := rest.RESTClientFor(config)
+		framework.ExpectNoError(err)
+
+		httpClient := restClient.Client
+
+		ginkgo.By("Making sure no previous rollout is in progress")
+		clusterApiServer, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Get(context.Background(), "cluster", metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		gomega.Expect(clusterApiServer.Status.NodeStatuses[0].TargetRevision).To(gomega.Equal(int32(0)))
+
+		ginkgo.By("Forcing API rollout")
+		forceApiRollout(oc)
+
+		ginkgo.By("Expecting API to become unavailable")
+		wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+			ready, _, err := isApiReady(config, httpClient)
+			return !ready, err
+		})
+
+		start := time.Now()
+
+		ginkgo.By("Expecting API to become ready")
+		wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+			ready, _, _ := isApiReady(config, httpClient)
+			return ready, nil
+		})
+
+		end := time.Now()
+
+		ginkgo.By("Measuring disruption duration time")
+		gomega.Expect(end.Sub(start)).To(gomega.BeNumerically("<", 40*time.Second))
```

Comment:
why 40? We said 60, didn't we?

Reply:
@romfreiman any preference on that one?
We can choose 60s as the original SLA or 40s to track regressions

Reply:
where are we today on average and on 99% quantile? I can see to put the 99% value here if it is very clear that the SLO is 60s, but we put a lower number to not regress, both in go doc and in test output.

Reply:
We're way better from 40 seconds. Most of the time it's between 7-10 seconds, the maximal time I've got is 11 seconds in a couple of runs that I've tried

Reply:
Let me update the message real quick